24.10.0-rc1 unable to find kernel modules from downloads.openwrt.org

### Describe the bug

This bug potentially affects **ALL** future point/tagged releases.

Kernel modules are now put only in `kmods` and not in `packages` under `targets`, which is nice.

We can compare output from different branches and point releases:

- https://downloads.openwrt.org/snapshots/targets/x86/64/packages/
- https://downloads.openwrt.org/releases/24.10-SNAPSHOT/targets/x86/64/packages/
- https://downloads.openwrt.org/releases/24.10.0-rc1/targets/x86/64/packages/
- https://downloads.openwrt.org/releases/23.05-SNAPSHOT/targets/x86/64/packages/
- https://downloads.openwrt.org/releases/23.05.5/targets/x86/64/packages/

However, `FeedSourcesAppendOPKG`, `FeedSourcesAppendAPK`, and `FeedSourcesAppend` only append `kmods` feed to

1. `/etc/opkg/distfeeds.conf` (OPKG)
2. `/etc/apk/repositories.d/distfeeds.list` (APK)
3. ImageBuilder `repositories.conf` (OPKG)
4. ImageBuilder `repositories` (APK)

for snapshot builds:

- https://github.com/openwrt/openwrt/blob/main/include/feeds.mk#L35
- https://github.com/openwrt/openwrt/blob/main/include/feeds.mk#L49
- https://github.com/openwrt/openwrt/blob/openwrt-24.10/include/feeds.mk#L35
- https://github.com/openwrt/openwrt/blob/openwrt-24.10/include/feeds.mk#L49
- https://github.com/openwrt/openwrt/blob/openwrt-23.05/include/feeds.mk#L31

Thus new point releases like 24.10.0-rc1 won't be able to find kernel modules from downloads.openwrt.org.


### OpenWrt version

r28158-d276b4c91a

### OpenWrt release

24.10.0-rc1

### OpenWrt target/subtarget

x86/64

### Device

QEMU Standard PC (Q35 + ICH9, 2009)

### Image kind

Official downloaded image

### Steps to reproduce

- download 24.10.0-rc1 ImageBuilder: https://downloads.openwrt.org/releases/24.10.0-rc1/targets/x86/64/openwrt-imagebuilder-24.10.0-rc1-x86-64.Linux-x86_64.tar.zst
- build image: `make image PROFILE=generic`

### Actual behaviour

```log
Collected errors:
 * pkg_hash_check_unresolved: cannot find dependency kmod-nf-conntrack-netlink for libnetfilter-conntrack3
 * pkg_hash_fetch_best_installation_candidate: Packages for libnetfilter-conntrack3 found, but incompatible with the architectures configured
 * pkg_hash_check_unresolved: cannot find dependency kmod-nft-core for nftables-json
 * pkg_hash_fetch_best_installation_candidate: Packages for nftables-json found, but incompatible with the architectures configured
 * pkg_hash_check_unresolved: cannot find dependency kmod-nft-core for firewall4
 * pkg_hash_check_unresolved: cannot find dependency kmod-nft-fib for firewall4
 * pkg_hash_check_unresolved: cannot find dependency kmod-nft-offload for firewall4
 * pkg_hash_check_unresolved: cannot find dependency kmod-nft-nat for firewall4
 * pkg_hash_fetch_best_installation_candidate: Packages for firewall4 found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package firewall4.
 * opkg_install_cmd: Cannot install package kmod-amazon-ena.
 * opkg_install_cmd: Cannot install package kmod-amd-xgbe.
 * opkg_install_cmd: Cannot install package kmod-bnx2.
 * opkg_install_cmd: Cannot install package kmod-button-hotplug.
 * opkg_install_cmd: Cannot install package kmod-dwmac-intel.
 * opkg_install_cmd: Cannot install package kmod-e1000.
 * opkg_install_cmd: Cannot install package kmod-e1000e.
 * opkg_install_cmd: Cannot install package kmod-forcedeth.
 * opkg_install_cmd: Cannot install package kmod-fs-vfat.
 * opkg_install_cmd: Cannot install package kmod-igb.
 * opkg_install_cmd: Cannot install package kmod-igc.
 * opkg_install_cmd: Cannot install package kmod-ixgbe.
 * opkg_install_cmd: Cannot install package kmod-nft-offload.
 * opkg_install_cmd: Cannot install package kmod-r8169.
 * opkg_install_cmd: Cannot install package kmod-tg3.
 * pkg_hash_check_unresolved: cannot find dependency kmod-nft-core for nftables-nojson
 * satisfy_dependencies_for: Cannot satisfy the following dependencies for nftables-json:
 *      kmod-nft-core
 * opkg_install_cmd: Cannot install package nftables.
 * pkg_hash_check_unresolved: cannot find dependency kmod-ppp for ppp
 * pkg_hash_fetch_best_installation_candidate: Packages for ppp found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package ppp.
 * pkg_hash_check_unresolved: cannot find dependency kmod-pppoe for ppp-mod-pppoe
 * pkg_hash_fetch_best_installation_candidate: Packages for ppp-mod-pppoe found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package ppp-mod-pppoe.
make[2]: *** [Makefile:228: package_install] Error 255
make[1]: *** [Makefile:155: _call_image] Error 2
make: *** [Makefile:331: image] Error 2
```

### Expected behaviour

Successful build

### Additional info

_No response_

### Diffconfig

_No response_

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

--- a/include/feeds.mk
+++ b/include/feeds.mk
@@ -37,7 +37,7 @@ define FeedSourcesAppendOPKG
   echo 'src/gz %d_core %U/targets/%S/packages'; \
   $(strip $(if $(CONFIG_PER_FEED_REPO), \
 	echo 'src/gz %d_base %U/packages/%A/base'; \
-	$(if $(filter %SNAPSHOT-y,$(VERSION_NUMBER)-$(CONFIG_BUILDBOT)), \
+	$(if $(CONFIG_BUILDBOT), \
 		echo 'src/gz %d_kmods %U/targets/%S/kmods/$(LINUX_VERSION)-$(LINUX_RELEASE)-$(LINUX_VERMAGIC)';) \
 	$(foreach feed,$(FEEDS_AVAILABLE), \
 		$(if $(CONFIG_FEED_$(feed)), \
@@ -51,7 +51,7 @@ define FeedSourcesAppendAPK
   echo '%U/targets/%S/packages/packages.adb'; \
   $(strip $(if $(CONFIG_PER_FEED_REPO), \
 	echo '%U/packages/%A/base/packages.adb'; \
-	$(if $(filter %SNAPSHOT-y,$(VERSION_NUMBER)-$(CONFIG_BUILDBOT)), \
+	$(if $(CONFIG_BUILDBOT), \
 		echo '%U/targets/%S/kmods/$(LINUX_VERSION)-$(LINUX_RELEASE)-$(LINUX_VERMAGIC)/packages.adb';) \
 	$(foreach feed,$(FEEDS_AVAILABLE), \
 		$(if $(CONFIG_FEED_$(feed)), \

